File: Basic/bjgood01/Queue/18258.py
from collections import deque
import sys
queue = deque()
n = int(sys.stdin.readline().rstrip())
for i in range(n):
    s = list(sys.stdin.readline().rstrip().split())
    if s[0] == 'push':
        queue.append(int(s[1]))
    elif s[0] == 'front':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[0])
    elif s[0] == 'back':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])
    elif s[0] == 'size':
        print(len(queue))
    elif s[0] == 'empty':
        if len(queue) == 0:
            print(1)
        else:
            print(0)
    elif s[0] == 'pop':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.popleft())

File: Basic/dlgustj206/Queue/18258.cpp
#include <iostream>
#include <queue>
using namespace std;

int main(void)
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin>>n;
    queue<int> q;

    while(n--){
        string s;
        cin>>s;
        if(s=="push"){
            int x;
            cin>>x;
            q.push(x);
        }
        else if(s=="pop"){
            if(q.empty()) cout<<-1<<"\n";
            else{
                cout<<q.front()<<"\n";
                q.pop();
            }
        }
        else if(s=="size") cout<<q.size()<<"\n";
        else if(s=="empty"){
            if(q.empty()) cout<<1<<"\n";
            else cout<<0<<"\n";
        }
        else if(s=="front"){
            if(q.empty()) cout<<-1<<"\n";
            else cout<<q.front()<<"\n";
        }
        else if(s=="back"){
            if(q.empty()) cout<<-1<<"\n";
            else cout<<q.back()<<"\n";
        }
    }
}